src/Datadog.Trace.ClrProfiler.Managed/Integrations/AspNetWebApi2Integration.cs

```diff
@@ -58,72 +61,98 @@ private static async Task<HttpResponseMessage> ExecuteAsyncInternal(object apiCo
             {
                 try
                 {
+                    // call the original method, inspecting (but not catching) any unhandled exceptions
                     var responseMessage = await executeAsyncFunc(apiController, controllerContext, cancellationToken).ConfigureAwait(false);
 
-                    // some fields aren't set till after execution, so populate anything missing
-                    UpdateSpan(controllerContext, scope.Span);
+                    if (scope != null)
+                    {
+                        // some fields aren't set till after execution, so populate anything missing
+                        UpdateSpan(controllerContext, scope.Span);
+                    }
 
                     return responseMessage;
                 }
-                catch (Exception ex)
+                catch (Exception ex) when (scope?.Span.SetExceptionForFilter(ex) ?? false)
                 {
-                    scope.Span.SetException(ex);
+                    // unreachable code
                     throw;
                 }
             }
         }
 
-        private static Scope CreateScope(object controllerContext)
-        {
-            var scope = Tracer.Instance.StartActive(OperationName);
-            UpdateSpan(controllerContext, scope.Span);
-            return scope;
-        }
-
-        private static void UpdateSpan(dynamic controllerContext, Span span)
+        private static Scope CreateScope(dynamic controllerContext)
```

Comment:
I'm trying to merge down from develop at the moment, this shouldn't have made it through a PR...if the intent here was to close the disposal leak/gap, it doesn't do it (and needs to).  I'll likely get it in my merge, but it's gonna take a bit given the churn here...

Reply:
Where's the leak? `CreateScope()` will always return the `Scope` if it's created (or `null` otherwise) and the `using` statement will dispose it.

Anyway, this is not critical at the moment. It's always been this way without issues. I agree that the leak should be fixed, but it should be done in it's own PR.

Reply:
Maybe it does actually close the gap now that I'm getting into the merge, so ignore this for now